digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/assetUpdater/ChangeReporter.scala

```diff
@@ -368,19 +368,22 @@ object ChangeReporter {
     (stringWriter.toString, contentRows)
   }
 
-  def saveReportToS3(assetName: String, body: String, contentRowCount: Int, hasGeometry: Boolean = false) = {
+  def saveReportToS3(assetName: String, changesProcessedUntil: DateTime, body: String, contentRowCount: Int,
+                     hasGeometry: Boolean = false): Unit = {
     val date = DateTime.now().toString("YYYY-MM-dd")
+    val untilDate = changesProcessedUntil.toString("YYYY-MM-dd")
     val withGeometry = if (hasGeometry) "_withGeometry" else ""
-    val path = s"${date}/${assetName}_${date}_${contentRowCount}content_rows${withGeometry}.csv"
+    val path = s"$date/${assetName}_${untilDate}_${contentRowCount}content_rows$withGeometry.csv"
```

Comment:
tuli mieleen, olisikohan testauksen näkökulmasta parempi jos lisättäisiin versiointi S3 buckettiin, jos pitää prosessoida sama muutossetti useampaan kertaan esim. kun löytyy bugi koodista ja se korjataan.

Reply:
Nykyisellä tyylillä ei taitaisi toimia kun objektin nimen alku muodostuu samuutuksen ajon päivämäärästä (esim 2023-02-02/). Jos tiputettaisiin alkuosa pois niin sit versiointi olisi mahdollinen, mut en tiedä miten luettava tuosta bucketista tulee jos päivittäin tulee tietolajien määrä x2 määrä uusia raportteja ja kaikki on yhtenä listana. Piiloon taitaisi myös jäädä milloin mitäkin on ajettu (ellei katsota muokkaus päivämääriä) kun nyt yhden päivän aikana ajetut on ainakin näennäisesti omissa kansioissaan.

Reply:
Periaatteessa jos nimeäminen muutettaisiin esim s"${assetName}/${untilDate}_${contentRowCount}content_rows$withGeometry.csv" ja lisäis mukaan versioinnin niin sit kaikki olis tietolajien mukaan järjestettynä ja vois seurata paremmin miten eri samuutuksissa on käynyt. Kokonaiskuva siitä mitä raportteja on yhden päivän aikana tehty jäisi kuitenkin piiloon

Reply:
Mietin, siis että voitaisiin lisätä versiointi ja säilyttää nykyinen nimeäminen, tälleen voitaisiin vertailla samana päivänä ajettujen samuutusten tuloksia keskenään esim. bugi korjausta testatessa. Ei nyt ehkä akuutein asia

Reply:
S3 versiointi vaatii että objektien keyt on tismalleen samat. Ennen tätä muutosta objektin nimi koostui ajon pvm + assetin nimi + ajon pvm + raportoitavien kohteiden määrä (?), joten versiointi olis auttanut vain jos muokattujen kohteiden määrä olisi ollut tismalleen sama. Jos muutoksia olisi ollut eri määrä niin sit oltaisiin luotu kokonaan uusi objekti (esim 2023-06-09/Obstacles_2023-06-09_**50**content_rows.csv ja esim 2023-06-09/Obstacles_2023-06-09_**100**content_rows.csv). Uuden toteutuksen mukaan kaikki kirjoitetaan omina objekteina ja omasta mielestä niiden pitäisi myös järjestyä ihan nätisti vanhimmasta uusimpaan. Mut joo jos ajetaan joku samuutus samana päivänä uudestaan ja muutosrivejä tulee tismalleen sama määrä niin ylikirjoitetaan vanha versio.